express-post-json/index.js

```diff
@@ -0,0 +1,32 @@
+const express = require('express');
+const app = express();
+
+let nextId = 1;
+const grades = {};
+
+app.listen(3000, () => {
+  // eslint-disable-next-line
+  console.log('Express server listening on port 3000');
+});
+
+const gradesArr = [];
+app.get('/api/grades', function (req, res) {
+  res.json(gradesArr);
+});
+
+app.use(express.json());
+
+app.post('/api/grades', function (req, res) {
+  const gradeObj = {
+    course: req.body.course,
+    name: req.body.name,
+    score: req.body.score,
+    id: nextId
+  };
+  res.status(201).json(gradeObj);
+  grades.nextId = gradeObj;
+  nextId++;
+  for (const key in grades) {
+    gradesArr.push(grades[key]);
+  }
```

Comment:
Move this and your array declaration into the GET route. We want to store the grades in an _object_ data model.